src/engine/SCons/Action.py

```diff
@@ -477,6 +477,49 @@ def execute(self, target, source, env):
 
         cmd_list, ignore, silent = self.process(target, map(rfile, source), env)
 
+        # DLADD: JRH 20151022 SF37871/SF37872: DLE now contains to many .java and .cs files
+        #                                      to compile with the standard windows cmd.exe
+        # on windows, use powershell when available/necessary (to avoid command line length limit failures)
+        if (sys.platform == 'win32') and (cmd_line[0] == 'javac'):
+            try:
+                maxline = int(env.subst('$MAXLINELENGTH'))
+            except ValueError:
+                maxline = 2048
+
+            psc_key = None
+            try:
+                if sys.version_info < (3,0):
+                    import _winreg as wreg
+                else:
+                    import winreg as wreg
+
+                psc_key = wreg.OpenKey(wreg.HKEY_CLASSES_ROOT, 'Microsoft.PowerShellConsole.1')
+                psc_value = wreg.QueryValueEx(psc_key, 'FriendlyTypeName')
+
+                # the parse logic (below) might be a bit sketchy since it implies a schema
+                # for the registry key. if that schema should change, this logic will fail
+                # to tease the path out of the registry key's raw value. at present, the
+                # raw value is a tuple that contains a compound, "tuple-like" string.
+                raw_path = psc_value[0].split('"')[1]
+
+                if '%systemroot%' in raw_path:
+                    powershell_path = string.replace(raw_path, '%systemroot%', os.getenv('systemroot'))
+
+                if os.path.isfile(powershell_path) == True:
```

Comment:
IMO, this is a bad habit - testing a value for EQUALITY to a boolean, what's more it also violates a python style guide that says "always test singletons with identity"  i.e. X is None.